src/map.rs

```diff
@@ -241,58 +252,89 @@ pub fn process_loaded_tile_maps(
-
-                        let has_new = (0..(columns * rows) as u32).fold(false, |total, next| {
-                            total || !texture_atlas_map.contains_key(&(tileset.first_gid + next))
-                        });
-                        if has_new {
-                            let atlas = TextureAtlas::from_grid(
-                                texture_handle.clone(),
-                                Vec2::new(tile_width, tile_height),
-                                columns,
-                                rows,
-                            );
-                            let atlas_handle = texture_atlases.add(atlas);
-                            for i in 0..(columns * rows) as u32 {
-                                if texture_atlas_map.contains_key(&(tileset.first_gid + i)) {
-                                    continue;
-                                }
-                                // println!("insert: {}", tileset.first_gid + i);
-                                texture_atlas_map
-                                    .insert(tileset.first_gid + i, atlas_handle.clone());
-                            }
+                    if !object_gids.contains(&Some(tileset.first_gid)) { continue; }
+
+                    // For simplicity use textureAtlasSprite for object layers
+                    // these insertions should be limited to sprites referenced by objects
+                    let tile_width = tileset.tile_width as f32;
+                    let tile_height = tileset.tile_height as f32;
+                    let texture_width = image.width as f32;
+                    let texture_height = image.height as f32;
+                    let columns = (texture_width / tile_width).floor() as usize;
+                    let rows = (texture_height / tile_height).floor() as usize;
+
+                    let has_new = (0..(columns * rows) as u32).fold(false, |total, next| {
+                        total || !texture_atlas_map.contains_key(&(tileset.first_gid + next))
+                    });
+                    if !has_new { continue; }
+
+                    let atlas = TextureAtlas::from_grid(
+                        texture_handle.clone(),
+                        Vec2::new(tile_width, tile_height),
+                        columns,
+                        rows,
+                    );
+                    let atlas_handle = texture_atlases.add(atlas);
+                    for i in 0..(columns * rows) as u32 {
+                        if texture_atlas_map.contains_key(&(tileset.first_gid + i)) {
+                            continue;
+                        }
+                        // println!("insert: {}", tileset.first_gid + i);
+                        texture_atlas_map
+                            .insert(tileset.first_gid + i, atlas_handle.clone());
+                    }
+                } else if let Some(_) = tileset.tiles.first().and_then(|tile| tile.images.first()) {
```

Comment:
If there's no image on the tileset, here we next check if the first tile has an individual image.

I'm assuming here that either *every* tile in a tileset will have an individual image, or *none* of them will. I think that's right, but maybe there's a more complex Tiled arrangement I'm not aware of?

Reply:
No, that should be a safe assumption.